relay/nakama/receipt/dispatcher.go

```diff
@@ -32,33 +32,34 @@ type Receipt struct {
 // Dispatcher continually polls Cardinal for transaction receipts and dispatches them to any subscribed
 // channels. The subscribed channels are stored in the sync.Map.
 type Dispatcher struct {
-	ch chan *Receipt
+	ch chan []*Receipt
```

Comment:
Rather then making a channel of an array of receipts you can make a buffered channel. A buffered channel will not block until the buffer is full. 

You mentioned "clog" in your PR description. Did you mean deadlock? Can you explain how the deadlock occurs? Looks like as long as Dispatch is running as a go routine it can never deadlock. 

It looks to me there can only be dropped messages. Is this what you mean by "clogged". What were you seeing here?

Reply:
I can explain, Jeremy and I talked about this one and decided to go with batched (a slice of) receipts here because if you think of Cardinal as the producer, and the Dispatcher as the middle man, and the Notifier as the producer, what I was seeing was that the consumer was not always able to take off receipts from the channel as fast as the producer was putting them on.

Buffering them is what I first thought of but we cannot know how many receipts Cardinal is going to send over in a tick and if that number ever exceeds the buffered size we are going to drop receipts. That's why @jerargus and I agreed that using a slice of receipts in this case works better since no receipts will be dropped.

Reply:
@namrapatel I talked to jeremy. To summarize, 

He told me the main purpose of the slice as opposed to a buffered channel here was to use NotificationsSend as opposed NotificationSend (without the s) such that it can operate on a slice.

He also stated that the purpose for dispatcher for receipts purpose was to have multiple subscribers on the Producer for the future and he doesn't want one subscriber to potentially block it. 

I can agree with the two reasons above.